data/core/rootview.lua

```diff
@@ -142,7 +142,9 @@ function Node:remove_view(root, view)
     local parent = self:get_parent_node(root)
     local is_a = (parent.a == self)
     local other = parent[is_a and "b" or "a"]
-    if other:get_locked_size() then
+    local locked_size_x, locked_size_y = other:get_locked_size()
+    if self.is_primary_node
+       or (parent.type == "hsplit" and locked_size_x or locked_size_y) then
```

Comment:
The logical expression between parenthesis looks wrong because if parent.type == "split" and locked_size_x is false it will return locked_size_y. I guess this is not what you want.

Reply:
No, if `parent.type` is `"hsplit"`, `locked_size_x` is returned, else (only other possible value is `"vsplit"`) `locked_size_y` is returned.

Reply:
Nope, see: https://www.lua.org/manual/5.2/manual.html#3.4.4

In Lua the expression `a and b or c` is equivalent to `(a and b) or c`. If a is true and b is false it will return c.

For example if you write:

```lua
local a, b, c = true, false, "boum"
print(a and b or c)
```

it will print "BOUM" and not "false" as you may expect.

This is a common pitfall. People assume that (a and b or c) behaves like the C ternary operator (a ? b : c). In reality it actually behaves like the ternary operator except when b evaluate to a falsy value where it returns c instead of b.

Reply:
Oh, I see. I'll fix that later then.